src/provider/model/Model.ts

```diff
@@ -208,6 +215,13 @@ export class Model {
         }
     }
 
+    private async onApiHandlerConnection(identity: Identity): Promise<void> {
+        if (await this._environment.getEntityType(identity) === EntityType.EXTERNAL_CONNECTION) {
```

Comment:
Since we won't ever get a window closed event for .NET connections (right?), de-register on disconnect?

And I'm not convinced things will work right with the expect system? E.g., we don't get a seen event for  these windows, and the close promise will never reject

Reply:
(And if you're fiddling around there, expect merge conflicts - sorry!)

Reply:
I'll defer to the second phase of this story, if there's no objections?

I'd want to find a way to store the entity type on the "window" object itself, so that we can keep the disconnect handler synchronous. Since some key parts of this are in flux due to other PR, maybe makes sense for those changes to be finalised first?

Reply:
No objection, but put in TODOs